analyses/chromosomal-instability/02b-plot-chr-instability-by-histology.Rmd

```diff
@@ -0,0 +1,195 @@
+  warning("The parameter min_samples should be numeric. Trying to coerce to numeric.")
+  params$min_samples <- as.numeric(params$min_samples)
+} 
+```
+
+### Directories and Files
+
+```{r}
+# Path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# Path to output directory
+plots_dir <- "plots"
+
+# Path to tumor type plots output directory
+hist_plots_dir <- file.path(plots_dir, "tumor-type")
+
+# Create the hist_plots_dir  if it does not exist
+if (!dir.exists(hist_plots_dir)) {
+  dir.create(hist_plots_dir, recursive = TRUE)
+}
+```
+
+Here's all the input files we will need: 
+
+```{r}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+hist_binned_breaks_file <- file.path("breakpoint-data", "histology_breakpoint_binned_counts.RDS")
+total_densities_file <- file.path("breakpoint-data", "intersection_of_breaks_densities.tsv")
+```
+
+Output plot: 
+
+```{r}
+cdf_plot_file <- file.path(plots_dir, "breaks_cdf_plot.png")
+```
+
+### Read in data 
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(metadata_file)
+```
+
+Load in the localized by histology breakpoint data. 
+
+```{r}
+hist_break_local <- readr::read_rds(hist_binned_breaks_file)
+```
+
+Read in the total intersection breaks densities TSV that was made by the 
+`00-setup-breakpoint-data.R` script. 
+
+```{r}
+total_break_densities_df <- readr::read_tsv(
+  total_densities_file
+) %>%
+  # Add on the short histology column from the metadata for plotting purposes
+  dplyr::inner_join(metadata %>% dplyr::select(Kids_First_Biospecimen_ID, short_histology),
+    by = c("samples" = "Kids_First_Biospecimen_ID")
+  )
+```
+
+## Plot CDF plot by histology
+
+```{r}
+breaks_cdf <- total_break_densities_df  %>%
+  # Only plot histologies groups with more than `min_samples` number of samples
+  dplyr::group_by(short_histology, add = TRUE) %>%
+  dplyr::filter(dplyr::n() > params$min_samples) %>%
+  # Calculate histology group mean
+  dplyr::mutate(
+    hist_mean = mean(breaks_density),
+    hist_rank = rank(breaks_density) / dplyr::n()
+  ) %>%
+  dplyr::ungroup() %>%
+  # Now we will plot these as cummulative distribution plots
+  ggplot2::ggplot(ggplot2::aes(
+    x = hist_rank, 
+    y = breaks_density,
+    color = short_histology
+  )) +
+  ggplot2::geom_point() +
+  # Add summary line for mean
+  ggplot2::geom_hline(ggplot2::aes(yintercept = hist_mean)) +
+  # Separate by histology
+  ggplot2::facet_wrap(~short_histology, nrow = 1) +
+  ggplot2::theme_classic() +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Chromosomal Breaks per Mb") +
+  # Transform to log10 make non-log y-axis labels
+  ggplot2::scale_y_continuous(trans = "log10", breaks = c(0, 1, 10, 30)) +
+  ggplot2::theme(legend.position = "none") +
+  # Get rid of x-axis ticks
+  ggplot2::theme(
+    axis.text.x = ggplot2::element_blank(),
+    axis.ticks.x = ggplot2::element_blank(),
+    strip.text = ggplot2::element_text(size = 10)
+  )
```

Comment:
I did some editing here for a nicer plot, IMO. Main changes are strip.placement, strip.position, strip.background and axis.text.x.

```r
  ggplot2::ggplot(ggplot2::aes(
    x = hist_rank, 
    y = breaks_density,
    color = short_histology
  )) +
  ggplot2::geom_point() +
  # Add summary line for mean
  ggplot2::geom_segment(x = 0, xend = 1, color = "grey", 
                        ggplot2::aes( y = hist_mean, yend = hist_mean)) +
  # Separate by histology
  ggplot2::facet_wrap(~short_histology, nrow = 1, strip.position = "bottom") +
  ggplot2::theme_classic() +
  ggplot2::xlab("") +
  ggplot2::ylab("Chromosomal Breaks per Mb") +
  # Transform to log10 make non-log y-axis labels
  ggplot2::scale_y_continuous(trans = "log10", breaks = c(0, 1, 3, 10, 30)) +
  ggplot2::scale_x_continuous(limits = c(-0.1, 1.1), breaks = c()) +
  ggplot2::theme(legend.position = "none") +
  ggplot2::theme(
    axis.text.x = ggplot2::element_blank(),
    axis.ticks.x = ggplot2::element_blank(),
    strip.placement = "outside",
    strip.text = ggplot2::element_text(size = 10, angle = 90, hjust = 1), 
    strip.background = ggplot2::element_rect(fill = NA, color = NA)
  )
```